File: Python/triangle.py
def not_degenerate(sides):
    """
    Determine if a triangle is degenerate 
    
    Parameter sides: list of int - with the given sides, we can determine if a triangle is degenerate .
    """
    
    sides.sort()
    a,b,c = sides
    return ((a + b) > (c))

def size_not_0(sides):
    a,b,c = sides
    return a != 0 or b != 0 or c != 0

def equilateral(sides):
    """
    Determine if a triangle is equilateral
    
    Parameter sides: list of int - with the given sides, we can determine if a triangle is equilateral.
    """
    a,b,c = sides
    return size_not_0(sides) and a == b == c



def isosceles(sides):
    """
    Determine if a triangle is isosceles
    
    Parameter sides: list of int - with the given sides, we can determine if a triangle is isosceles.
    """
    
    
    a,b,c = sides
    
    return not_degenerate(sides) and size_not_0(sides) and (a == b or b == c or a == c)
        

def scalene(sides):
    """
    Determine if a triangle is scalene
    
    Parameter sides: list of int - with the given sides, we can determine if a triangle is scalene.
    """

    
    a,b,c = sides
    
    return not_degenerate(sides) and size_not_0(sides) and a != b != c != a
    

File: Ruby/triangle.rb
=begin
Write your code for the 'Triangle' exercise in this file. Make the tests in
`triangle_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/triangle` directory.
=end

class Triangle
    def initialize(distance)
      @sides = distance
      @longest = 0
      distance.each do |sides|
        if @longest < sides
          @longest = sides
        end
      end
    end
  
    def scalene?
      @sides[0] != @sides[1] and @sides[1] != @sides[2] and @sides[0] != @sides[2] and (@longest < (@sides.sum - @longest))
    end
  
    def isosceles?
     ( (@sides[0] == @sides[1]) or (@sides[0] == @sides[2]) or (@sides[2] == @sides[1])) and (@longest < (@sides.sum - @longest))
    end
  
    def equilateral?
      @sides[0] == @sides[1] and @sides[1] == @sides[2] and @sides[2] != 0
    end
  
      
  end